torch/distributions/geometric.py

```diff
@@ -0,0 +1,66 @@
+from numbers import Number
+
+import torch
+from torch.distributions import constraints
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all, probs_to_logits, logits_to_probs, lazy_property, _finfo
+from torch.nn.functional import binary_cross_entropy_with_logits
+
+
+class Geometric(Distribution):
+    r"""
+    Creates a Geometric distribution parameterized by `probs`. It represents the probability that in
+    k + 1 Bernoulli trials, the first k trials failed, before seeing a success
+
+    Samples are positive integers [1, inf).
```

Comment:
I'm happy that you've implemented the `{0,1,2,...}` version of Geometric rather than the `{1,2,3...}` version. Could you update docs to reflect this? This should say "nonnegative integers [0, inf)".

Reply:
Sounds reasonable. Actually the only use case I know of for constraints is in deriving unconstrained optimization spaces for optimizers in SVI, as in #64. I added these discrete constraints for consistency, but barring a use case we might merge all these into a single `discrete` constraint indicating that continuous optimization is not possible. Do you know of any use cases for discrete constraints?

Reply:
I was thinking about continuous variables, for example `probs` here has `unit_interval` constraint, but what I actually want is (0,1]

Reply:
For my use cases in SVI, I only care about constraints up to measure 0, so `(0,1) = [0,1]`. Both are mapped to the real line via `logit()`. Generally, optimizers should never hit a boundary.